### PR TITLE
Change update status to testing if every build is signed

### LINKED ISSUE #3475: [S] Message consumer: Move update from side-tag to testing
# Story:

As a user of multi-build gating,

I want a Bodhi message consumer to change the status of an update created for a side-tag to `testing` when all of its builds are signed by Robosignatory (i.e. tagged into the `*-testing` tag corresponding to its side-tag) and announce it via the message bus,

so that automated testing can commence.

# Background

pypingou's multi-build update workflow diagram \[[download](https://user-images.githubusercontent.com/820624/63262799-34940480-c287-11e9-9594-e177ce7e6d4d.png)\]:

This card corresponds to the "Move update to status: testing" step half-way down in the Bodhi column.

### PR DESCRIPTION
**Pull request by Zlopez**
Closes #3475

Signed-off-by: Michal Konečný <mkonecny@redhat.com>

### CODE REVIEW

**Comment by Zlopez** [comment]
This is waiting for #3481 to be merged.

**Comment by Zlopez** [comment]
I updated this PR with changes from #3481

File: bodhi/server/models.py [hunk 1]
```diff
@@ -2390,11 +2390,14 @@ def signed(self):
...
 
+        If the update is created ``from_tag`` always check if every build is signed.
+
         Returns:
             bool: ``True`` if the update is signed, ``False`` otherwise.
         """
-        if not self.release.pending_signing_tag:
-            return True
+        if not self.from_tag:
+            if not self.release.pending_signing_tag:
```
Comment (cverna) [inline bodhi/server/models.py]:
```suggestion
            if not self.release.pending_signing_tag and not self.from_tag:
```

File: bodhi/server/consumers/signed.py [hunk 2]
```diff
@@ -101,3 +107,10 @@ def __call__(self, message: fedora_messaging.api.Message):
             log.info("Build has been signed, marking")
             build.signed = True
             log.info("Build %s has been marked as signed" % build_nvr)
+
+            # If every build in update is signed change status to testing
+            if build.update.from_tag and build.update.signed():
+                log.info("Every build in update is signed, set status to testing")
+                build.update.status = UpdateStatus.testing
+                log.info("Update {} status has been set to testing".format(
```
Comment (cverna) [inline bodhi/server/consumers/signed.py]: Let's use f-strings :smile:

File: bodhi/server/consumers/signed.py [hunk 3]
```diff
@@ -101,3 +107,10 @@ def __call__(self, message: fedora_messaging.api.Message):
             log.info("Build has been signed, marking")
             build.signed = True
             log.info("Build %s has been marked as signed" % build_nvr)
+
+            # If every build in update is signed change status to testing
+            if build.update.from_tag and build.update.signed():
+                log.info("Every build in update is signed, set status to testing")
+                build.update.status = UpdateStatus.testing
```
Comment (cverna) [inline bodhi/server/consumers/signed.py:116]:
```suggestion
                from sqlalchemy import func
                
                build.update.status = UpdateStatus.testing
                build.update.date_testing = func.current_timestamp()
                build.update.request = None
```

**Comment by Zlopez** [comment]
I updated the PR with changes addressing @cverna comments

File: bodhi/server/consumers/signed.py [hunk 4]
```diff
@@ -101,3 +107,13 @@ def __call__(self, message: fedora_messaging.api.Message):
             log.info("Build has been signed, marking")
             build.signed = True
             log.info("Build %s has been marked as signed" % build_nvr)
+
+            # If every build in update is signed change status to testing
+            if build.update.from_tag and build.update.signed():
+                log.info("Every build in update is signed, set status to testing")
+                from sqlalchemy import func
```
Comment (cverna) [inline bodhi/server/consumers/signed.py]: Let's put this import with the rest of the imports at the top of the file :-)

Comment (Zlopez) [reply bodhi/server/consumers/signed.py]: Moved to the import section :-)

**Review by cverna — APPROVED** [review]
:+1:

**Pull request merged**